components/Link/src/index.scss

```diff
@@ -71,6 +71,11 @@
   width: var(--denhaag-link-icon-size);
 }
 
+li .denhaag-link--with-icon-start .denhaag-link__icon {
+  align-self: start;
```

Comment:
Uhh, this is to specific, since we also can align icons on the end of the line.

Therefor i would suggest again to replace the `align-items:` with `align-self` on line 64 of this file. The flex: 1 0 auto may also be appended to the `.denhaag-link__icon` class.